src/services/blocks/BlocksService.ts

```diff
@@ -576,4 +623,69 @@ export class BlocksService extends AbstractService {
 
 		return undefined;
 	}
+
+	/**
+	 * When querying a block this will immediately inform the request whether
+	 * or not the queired block is considered finalized at the time of querying.
+	 *
+	 * @param api ApiPromise to use for query
+	 * @param blockNumber Queried Block Number
+	 * @param queriedHash This is the Queried Hash param
+	 * @param finalizedHead This is the finalized head for our chain
+	 * @param checkFinalized If the passed in blockId is a hash we check query canonHash
+	 */
+	private async isFinalizedBlock(
+		api: ApiPromise,
+		blockNumber: Compact<BlockNumber>,
+		queriedHash: BlockHash,
+		finalizedHead: BlockHash,
+		checkFinalized: boolean
+	): Promise<boolean> {
+		if (checkFinalized) {
+			// The blockId url param is a hash
+			const [finalizedHeadBlock, canonHash] = await Promise.all([
+				// Returns a Finalized head Object
+				api.rpc.chain.getHeader(finalizedHead),
+				// Fetch the hash of the block with equal height on the canon chain.
+				// N.B. We assume when we query by number <= finalized head height,
+				// we will always get block on the finalized, canon chain.
+				api.rpc.chain.getBlockHash(blockNumber.unwrap()),
+			]);
+
+			// If queried by hash this is the original request param
+			const hash = queriedHash.toHex();
+
+			// If this conditional is satisfied, the queried hash is on a fork,
+			// and is not on the canonical chain and therefore not finalized
+			if (canonHash.toHex() !== hash) {
+				return false;
+			}
+
+			// Retreive the finalized head blockNumber
+			const finalizedHeadBlockNumber = finalizedHeadBlock?.number;
+
+			// If the finalized head blockNumber is undefined return false
+			if (!finalizedHeadBlockNumber) {
+				return false;
+			}
+
+			// Check if the finalized head blockNumber is greater than the
+			// blockNumber in the request. If so the requested block is finalized.
+			return blockNumber.unwrap().lte(finalizedHeadBlockNumber.unwrap());
+		} else {
+			// The blockId url param is an integer
+
+			// Returns a Finalized head Object
+			const finalizedHeadBlock = await api.rpc.chain.getHeader(
+				finalizedHead
+			);
+
+			// Retreive the finalized head blockNumber
+			const finalizedHeadBlockNumber = finalizedHeadBlock?.number;
```

Comment:
Should we check for undefined here for consistency with above? If am reading correctly I think we could also pull this logic out of the if/else since it is dupe with line 665

Reply:
Yea I agree, I just added that in. I overlooked that because of the typescript compiler. Good catch.